OpenBLAS 0.3.12 ARMv7 build fail

Newly released OpenBLAS 0.3.12 fails build for ARMv7 architecture

> make TARGET=CORTEXA9 USE_OPENMP=1

```
...
cc -c -O2 -DMAX_STACK_ALLOC=2048 -fopenmp -marm -Wall -DF_INTERFACE_GFORT -fPIC -DSMP_SERVER -DUSE_OPENMP -DNO_WARMUP -DMAX_CPU_NUMBER=4 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.12\" -mfpu=vfpv3 -march=armv7-a -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=sswap_k -DASMFNAME=sswap_k_ -DNAME=sswap_k_ -DCNAME=sswap_k -DCHAR_NAME=\"sswap_k_\" -DCHAR_CNAME=\"sswap_k\" -DNO_AFFINITY -I.. -UDOUBLE  -UCOMPLEX -UCOMPLEX -UDOUBLE ../kernel/arm/swap_vfp.S -o sswap_k.o
In file included from ../kernel/arm/../generic/../simd/intrin.h:56:0,
                 from ../kernel/arm/../generic/dot.c:30:
/usr/lib/gcc/arm-linux-gnueabihf/5/include/arm_neon.h:31:2: error: #error You must enable NEON instructions (e.g. -mfloat-abi=softfp -mfpu=neon) to use arm_neon.h
 #error You must enable NEON instructions (e.g. -mfloat-abi=softfp -mfpu=neon) to use arm_neon.h
  ^
In file included from ../kernel/arm/../generic/../simd/intrin.h:56:0,
                 from ../kernel/arm/../generic/dot.c:30:
/usr/lib/gcc/arm-linux-gnueabihf/5/include/arm_neon.h:31:2: error: #error You must enable NEON instructions (e.g. -mfloat-abi=softfp -mfpu=neon) to use arm_neon.h
 #error You must enable NEON instructions (e.g. -mfloat-abi=softfp -mfpu=neon) to use arm_neon.h
  ^
In file included from ../kernel/arm/../generic/../simd/intrin.h:69:0,
                 from ../kernel/arm/../generic/dot.c:30:
../kernel/arm/../generic/../simd/intrin_neon.h:10:9: error: unknown type name ‘float32x4_t’
 typedef float32x4_t v_f32;
         ^
../kernel/arm/../generic/../simd/intrin_neon.h: In function ‘v_muladd_f32’:
../kernel/arm/../generic/../simd/intrin_neon.h:32:14: warning: implicit declaration of function ‘vmlaq_f32’ [-Wimplicit-function-declaration]
     { return vmlaq_f32(c, a, b); }
              ^
../kernel/arm/../generic/../simd/intrin_neon.h: At top level:
../kernel/arm/../generic/../simd/intrin_neon.h:42:30: error: unknown type name ‘float32x4_t’
 BLAS_FINLINE float v_sum_f32(float32x4_t a)
                              ^
In file included from ../kernel/arm/../generic/../simd/intrin.h:69:0,
                 from ../kernel/arm/../generic/dot.c:30:
../kernel/arm/../generic/../simd/intrin_neon.h:10:9: error: unknown type name ‘float32x4_t’
 typedef float32x4_t v_f32;
         ^
../kernel/arm/../generic/../simd/intrin_neon.h: In function ‘v_muladd_f32’:
../kernel/arm/../generic/../simd/intrin_neon.h:32:14: warning: implicit declaration of function ‘vmlaq_f32’ [-Wimplicit-function-declaration]
     { return vmlaq_f32(c, a, b); }
              ^
../kernel/arm/../generic/../simd/intrin_neon.h: At top level:
../kernel/arm/../generic/../simd/intrin_neon.h:42:30: error: unknown type name ‘float32x4_t’
 BLAS_FINLINE float v_sum_f32(float32x4_t a)
                              ^
Makefile.L1:782: recipe for target 'sdsdot_k.o' failed
make[1]: *** [sdsdot_k.o] Error 1
make[1]: *** Waiting for unfinished jobs....
Makefile.L1:779: recipe for target 'dsdot_k.o' failed
make[1]: *** [dsdot_k.o] Error 1
make[1]: Leaving directory '/home/user/Downloads/OpenBLAS-0.3.12/kernel'
Makefile:164: recipe for target 'libs' failed
make: *** [libs] Error 1
```

--- a/Makefile.arm
+++ b/Makefile.arm
@@ -12,3 +12,8 @@ ifeq ($(CORE), ARMV6)
 CCOMMON_OPT += -mfpu=vfp
 FCOMMON_OPT += -mfpu=vfp
 endif
+
+ifdef HAVE_NEON
+CCOMMON_OPT += -mfpu=neon
+FCOMMON_OPT += -mfpu=neon
+endif

--- a/getarch.c
+++ b/getarch.c
@@ -1405,8 +1405,41 @@ int main(int argc, char *argv[]){
 
     printf("NUM_CORES=%d\n", get_num_cores());
 
-#if defined(__arm__) && !defined(FORCE)
+#if defined(__arm__) 
+#if !defined(FORCE)
+    fprintf(stderr,"get features!\n");
         get_features();
+#else
+    fprintf(stderr,"split archconfig!\n");
+    sprintf(buffer, "%s", ARCHCONFIG);
+
+    p = &buffer[0];
+
+    while (*p) {
+      if ((*p == '-') && (*(p + 1) == 'D')) {
+	p += 2;
+        if (*p != 'H') {
+		while( (*p != ' ') && (*p != '-') && (*p != '\0') && (*p != '\n')) {p++; }
+		if (*p == '-') continue;
+	}
+	while ((*p != ' ') && (*p != '\0')) {
+
+	  if (*p == '=') {
+	    printf("=");
+	    p ++;
+	    while ((*p != ' ') && (*p != '\0')) {
+	      printf("%c", *p);
+	      p ++;
+	    }
+	  } else {
+	    printf("%c", *p);
+	    p ++;
+	    if ((*p == ' ') || (*p =='\0')) printf("=1\n");
+	  }
+	}
+      } else p ++;
+    }
+#endif
 #endif
 
 

